internal/buildkite/access_token.go

```diff
@@ -0,0 +1,40 @@
+package buildkite
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/buildkite/go-buildkite/v4"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/rs/zerolog/log"
+)
+
+type AccessTokenClient interface {
+	Get(ctx context.Context) (buildkite.AccessToken, *buildkite.Response, error)
+}
+
+func AccessToken(ctx context.Context, client AccessTokenClient) (tool mcp.Tool, handler server.ToolHandlerFunc) {
+	return mcp.NewTool("access_token",
+			mcp.WithDescription("Get the details for the API access token that was used to authenticate the request"),
```

Comment:
As we play with it a bit more, we'll probably want to add some more info to these to hint to the model _why_ it would want to check the access token etc

Reply:
@catkins yeah I was reading through the docs resources seem to be more the direction we want to go https://modelcontextprotocol.io/docs/concepts/resources 

Some experimentation required.